### PR TITLE
Fix reading auth0_branding when enable_custom_domain_in_emails is true

### LINKED ISSUE #380: auth0_branding fails with an error
### Description

When added `auth0_branding` resource, got an error saying `Error: 404 Not Found: Template does not exist.`

### Terraform Version

```
Terraform v0.14.9
+ provider registry.terraform.io/alexkappa/auth0 v0.21.0
+ provider registry.terraform.io/hashicorp/local v2.0.0

Your version of Terraform is out of date! The latest version
is 0.15.0. You can update by downloading from https://www.terraform.io/downloads.html

```

### Affected Resource(s)

* auth0_branding

### Terraform Configuration Files

```hcl
# This was already existing with 0.20.0 version and imported using terraform import
resource "auth0_tenant" "tenant" {
  friendly_name   = "My Tenant"
  enabled_locales = ["en"]
  flags {
    enable_public_signup_user_exists_error = true
    enable_custom_domain_in_emails         = true
  }
  universal_login {
    colors {
      primary         = "#0059d6"
      page_background = "#000000"
    }
  }
}

# this is also an existing resource
resource "auth0_prompt" "prompt" {
  universal_login_experience = "new"
}

resource "auth0_branding" "my-brand" {
  logo_url = "<my_image_link>"
  colors {
    primary = "#0059d6"
    page_background = "#000000"
  }
}
```

terraform plan showed below changes

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # auth0_branding.transformco-brand will be created
  + resource "auth0_branding" "my-brand" {
      + favicon_url = (known after apply)
      + id          = (known after apply)
      + logo_url    = "<my_image_url>"

      + colors {
          + page_background = "#000000"
          + primary         = "#0059d6"
        }
    }

```

Once applied above changes, the changes got reflected in tenant. But, failed with error mentioned above.

### Expected Behavior

Resource should be created without any error.

### Actual Behavior

The resource got created and I see the updated values under my tenant! But, failed with below error

```
Error: 404 Not Found: Template does not exist.
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

- Using custom domain with tenant
- As shown colors were already defined under `auth0_tenant` resource.
- Using new universal login

### References

* #0000

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by nekootoko3**
### Proposed Changes

* Enables you to set auth0_branding when custom domain is enabled and universal_login is not set
	* By current provider behavior, when custom domain is enabled, custom universal login pages must be set. From [document](https://auth0.com/docs/brand-and-customize/universal-login-page-templates), when custom domain is enabled, it is true that universal login pages can be set, but you don't have to.

Fixes #380 

#### Acceptance Test Output

```bash
$ make testacc TESTS=TestAccBranding                                                         
==> Checking that code complies with gofmt requirements...
?       github.com/alexkappa/terraform-provider-auth0   [no test files]
=== RUN   TestAccBranding
--- PASS: TestAccBranding (8.76s)
PASS
coverage: 7.6% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0     9.120s  coverage: 7.6% of statements
?       github.com/alexkappa/terraform-provider-auth0/auth0/internal/debug      [no test files]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/random     0.423s  coverage: 0.0% of statements [no tests to run]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/validation 0.478s  coverage: 0.0% of statements [no tests to run]
?       github.com/alexkappa/terraform-provider-auth0/version   [no test files]
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by nekootoko3** [comment]
Testing in an environment with no custom domain is complete.
But I could not test in an enviroment with custom domain because I don't have any paid enviroment which I can test in.

File: auth0/resource_auth0_branding.go [hunk 1]
```diff
@@ -112,12 +111,10 @@ func readBranding(d *schema.ResourceData, m interface{}) error {
 	}
 
 	if t.Flags.EnableCustomDomainInEmails != nil && *t.Flags.EnableCustomDomainInEmails {
-		ul, err := api.Branding.UniversalLogin()
-		if err != nil {
```
Comment (nekootoko3) [inline auth0/resource_auth0_branding.go:116]: in the case of `t.Flags.EnableCustomDomainInEmails == true` and UniversalLogin is not set, the response is `404 not found` and branding cannot be updated.

**Comment by nekootoko3** [comment]
@alexkappa @mpalmer To merge this PR, is there anything for me to do?

**Comment by sergiughf** [comment]
Hey @nekootoko3! Thanks a lot for this great contribution! 

I took a look at it and pushed some small adjustments (mostly styling) and rebased on top of master.

This was a really useful fix!

**Pull request merged**